lang/core/os/is_virtual.go

```diff
@@ -0,0 +1,160 @@
+// Written by James Shubin <james@shubin.ca> and the project contributors
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+// Additional permission under GNU GPL version 3 section 7
+//
+// If you modify this program, or any covered work, by linking or combining it
+// with embedded mcl code and modules (and that the embedded mcl code and
+// modules which link with this program, contain a copy of their source code in
+// the authoritative form) containing parts covered by the terms of any other
+// license, the licensors of this program grant you additional permission to
+// convey the resulting work. Furthermore, the licensors of this program grant
+// the original author, James Shubin, additional permission to update this
+// additional permission if he deems it necessary to achieve the goals of this
+// additional permission.
+
+package coreos
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"runtime"
+	"strings"
+
+	"github.com/purpleidea/mgmt/lang/funcs/simple"
+	"github.com/purpleidea/mgmt/lang/types"
+	"github.com/purpleidea/mgmt/util/errwrap"
+)
+
+// The Desktop Management Interface (DMI) is a framwork for exposing system data
+// to software. We will make use of it to determine whether we're using a
+// virtualization platform. dmiFilesSlice is a slice of file paths which might
+// contain information related to the presence of a virtualization platform.
+// This is where we might find the values present in virtualizationVendorSlice.
+var dmiFilesSlice = []string{
+	"/sys/class/dmi/id/product_name",
+	"/sys/class/dmi/id/sys_vendor",
+	"/sys/class/dmi/id/board_vendor",
+	"/sys/class/dmi/id/bios_vendor",
+	"/sys/class/dmi/id/product_version",
+}
+
+// virtualizationVendorSlice is a slice of strings that might be found on DMI
+// related files (dmiFilesSlice) during the checks performed for the presence of
+// virtualization.
+var virtualizationVendorSlice = []string{
+	"KVM",
+	"OpenStack",
+	"KubeVirt",
+	"Amazon EC2",
+	"QEMU",
+	"VMware",
+	"VMW",
+	"innotek GmbH",
+	"VirtualBox",
+	"Oracle Corporation",
+	"Xen",
+	"Bochs",
+	"Parallels",
+	"BHYVE",
+	"Hyper-V",
+	"Apple Virtualization",
+	"Google Computer Engine",
+}
+
+func init() {
+	simple.ModuleRegister(ModuleName, "is_virtual", &simple.Scaffold{
+		T: types.NewType("func() bool"),
+		F: IsVirtual,
+	})
+}
+
+// IsVirtual is a shim for the isVirtual function,
```

Comment:
This is the doc string which will automatically get used in the documentation. So this one should be good =D